instant-distance-py/src/lib.rs

```diff
@@ -348,65 +427,50 @@ impl Neighbor {
 
 #[repr(align(32))]
 #[derive(Clone, Deserialize, Serialize)]
-struct FloatArray(#[serde(with = "BigArray")] [f32; DIMENSIONS]);
+struct FloatArray<M> {
+    array: Vec<f32>,
+    // TODO: we don't seem to give the point coordinates back maybe we can overwrite array with preprocessed value?
+    preprocessed: Option<Vec<f32>>,
+    phantom: PhantomData<M>,
```

Comment:
The TODO here seems like a major point. I think on real-world uses performance for this code would be much worse than the previous version?

Reply:
Yeah, on a microbenchmark it's not really visible but it's because tiny:
```
PYTHONPATH=instant-distance-py/test/ python3 -m timeit -n 10 -s 'import random, instant_distance; points = [[random.random() for _ in range(300)] for _ in range(1024)]; config = instant_distance.Config()' 'instant_distance.Hnsw.build(points, config)'
10 loops, best of 5: 198 msec per loop
PYTHONPATH=instant-distance-py/test/ python3 -m timeit -n 10 -s 'import random, instant_distance; points = [[random.random() for _ in range(300)] for _ in range(1024)]; config = instant_distance.Config(); config.distance_metric = instant_distance.DistanceMetric.Cosine' 'instant_distance.Hnsw.build(points, config)'
10 loops, best of 5: 199 msec per loop
```
On bigger datasets the extra allocations and memory would be noticeable. I'm changing this to make preprocessing in-place.